network/framework/src/application/metadata.rs

```diff
@@ -87,6 +88,14 @@ impl PeerMetadata {
 
     /// Returns a copy of the peer monitoring metadata
     pub fn get_peer_monitoring_metadata(&self) -> PeerMonitoringMetadata {
```

Comment:
can we remove this expensive function? and even remove the Clone from the struct?

Reply:
I did, but I did not remove clone because it's appears necessary in some places.